crates/core_simd/src/ops/assign.rs

```diff
@@ -0,0 +1,131 @@
+use super::*;
+use core::ops::{AddAssign, MulAssign}; // commutative binary op-assignment
+use core::ops::{BitAndAssign, BitOrAssign, BitXorAssign}; // commutative bit binary op-assignment
+use core::ops::{DivAssign, RemAssign, SubAssign}; // non-commutative binary op-assignment
+use core::ops::{ShlAssign, ShrAssign}; // non-commutative bit binary op-assignment
+
+// Arithmetic
+
+impl<T, U, const LANES: usize> AddAssign<U> for Simd<T, LANES>
```

Comment:
So this is fine, but since all of these traits are basically identical in form it might be worth using a macro?

Reply:
Sure. I just tried to uh, aggressively minimize macro usage (or at least, use more specialized macros).